### PR TITLE
Add accuracy tests for precise models lacking them

### PR DESCRIPTION
**Pull request by stinebuu**
With @jougs PR #2294 I remembered that I started working on issue #1493, extending accuracy testing for precise models, a long time ago. This PR fixes the first part of that issue.

I have added `iaf_psc_exp_ps` and `iaf_psc_exp_ps_lossless` to `test_iaf_ps_dc_accuracy.sli` and `test_iaf_ps_dc_t_accuracy.sli`. To make the test run, I had to update the models to use `expm1 + 1` instead of `exp` to increase accuracy.

It also seemed to me that the setting of `t_spike` for `iaf_psc_exp_ps_lossless` was wrong, so I have updated it to fit with the other models.

When it comes to the second part of issue #1493:
> These tests accuracy in the presence of DC current only, which is not a very relevant test case. Could one create tests with realistic spike input using arbitrary precision C++ or Python libraries to generate reference data?

I am not quite sure how to fix this, but I wanted to already create this PR for the first part of the issue, as it is good to also test the accuracy of `iaf_psc_exp_ps` and `iaf_psc_exp_ps_lossless`.

### CODE REVIEW

File: models/iaf_psc_exp_ps.cpp [hunk 1]
```diff
@@ -326,8 +326,11 @@ nest::iaf_psc_exp_ps::update( const Time& origin, const long from, const long to
       // update membrane potential
       if ( not S_.is_refractory_ )
       {
-        S_.y2_ =
-          V_.P20_ * ( P_.I_e_ + S_.y0_ ) + V_.P21_ex_ * S_.y1_ex_ + V_.P21_in_ * S_.y1_in_ + S_.y2_ * V_.exp_tau_m_;
+        // If we use S_.y2_ * std::exp( -V_.h_ms_ / P_.tau_m_ ) instead of
+        // V_.expm1_tau_m_ * S_.y2_ + S_.y2_ here, the accuracy decrease,
```
Comment (heplesser) [inline models/iaf_psc_exp_ps.cpp:330]:
```suggestion
        // V_.expm1_tau_m_ * S_.y2_ + S_.y2_ here, the accuracy decreases,
```

File: models/iaf_psc_exp_ps_lossless.cpp [hunk 2]
```diff
@@ -367,8 +366,11 @@ nest::iaf_psc_exp_ps_lossless::update( const Time& origin, const long from, cons
       // update membrane potential
       if ( not S_.is_refractory_ )
       {
+        // If we use S_.y2_ * std::exp( -V_.h_ms_ / P_.tau_m_ ) instead of
+        // V_.expm1_tau_m_ * S_.y2_ + S_.y2_ here, the accuracy decrease,
```
Comment (heplesser) [inline models/iaf_psc_exp_ps_lossless.cpp:370]:
```suggestion
        // V_.expm1_tau_m_ * S_.y2_ + S_.y2_ here, the accuracy decreases,
```

**Review by heplesser — APPROVED** [review]
Looks good except for two tiny comment typos, see suggestions.

File: testsuite/unittests/test_iaf_ps_dc_accuracy.sli [hunk 3]
```diff
@@ -167,23 +169,23 @@ params begin userdict begin
...
   {
-    (--------------------------) <-
+    (----------------------) <-
   } repeat
   endl
 
   exch
   {
-    { exch 24 setw exch <- (  ) <- } forall endl
+    { exch 20 setw exch <- (  ) <- } forall endl
```
Comment (clinssen) [inline testsuite/unittests/test_iaf_ps_dc_accuracy.sli:188]: What is this number?

Comment (heplesser) [reply testsuite/unittests/test_iaf_ps_dc_accuracy.sli:188]: The column width for tabular output.

**Review by clinssen — APPROVED** [review]
Cheers!

**Pull request merged**